actors/builtin/market/market_actor.go

```diff
@@ -534,21 +530,35 @@ func (a Actor) CronTick(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
 					rt.Abortf(exitcode.ErrIllegalState, "failed to delete pending proposal: %v", err)
 				}
 
-				if state.SectorStartEpoch == epochUndefined {
+				state, found, err := states.Get(dealID)
+				if err != nil {
+					rt.Abortf(exitcode.ErrIllegalState, "failed to get deal: %d", dealID)
+				}
+				// deal has been published but not activated yet -> terminate it as it has timed out
+				if !found {
 					// Not yet appeared in proven sector; check for timeout.
 					AssertMsg(rt.CurrEpoch() >= deal.StartEpoch, "if sector start is not set, we must be in a timed out state")
 
-					slashed := st.processDealInitTimedOut(rt, et, lt, dealID, deal, state)
+					slashed := st.processDealInitTimedOut(rt, et, lt, deal)
 					if !slashed.IsZero() {
 						amountSlashed = big.Add(amountSlashed, slashed)
 					}
 					if deal.VerifiedDeal {
 						timedOutVerifiedDeals = append(timedOutVerifiedDeals, deal)
 					}
+
+					if err := deleteDeal(dealID, states, proposals); err != nil {
```

Comment:
`deleteDeal` deletes from `states` but we know it's not there. That deletion will return an error. This should fail a test.

There are many ways around this, but I think simplest is probably to have separate `deleteDealProposal` and `deleteDealState` mutators, and perhaps a `deleteDealProposalAndState` for that common case. But actually since both of those primitives are direct wrappers of the ADT operations, maybe they don't deserve to be functions: just call directly when you only want one of them.

Reply:
Oh good catch. Fixed this.